DocumentDataAPI/DocumentDataAPI/Controllers/DocumentController.cs

```diff
@@ -0,0 +1,208 @@
+            {
+                List<DocumentModel> result = _repository.GetAll().ToList();
+                return result.Count == 0 ?
+                    NotFound() :
+                    Ok(result);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to fetch documents\n{e.Message}");
+                return Problem(e.Message);
+            }
+
+        }
+
+        /// <summary>
+        /// Retrieves a document based on the document id.
+        /// </summary>
+        /// <response code="200">Success: A document for the given id.</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("Get/{id:int?}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<DocumentModel> GetById(int id)
+        {
+            try
+            {
+                DocumentModel? result = _repository.Get(id);
+                return result == null ?
+                    NotFound() :
+                    Ok(result);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to fetch document with id: {id}.\n{e.Message}");
+                return Problem(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the number of documents in the database.
+        /// </summary>
+        /// <response code="200">Success: A number</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("GetTotalDocumentCount")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<int> GetTotalDocumentCount()
+        {
+            try
+            {
+                int result = _repository.GetTotalDocumentCount();
+                return Ok(result);
+            }
+            catch (Exception e)
+            {
+                return Problem(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a list of documents based on the source id.
+        /// </summary>
+        /// <response code="200">Success: A list of documents for the given source id.</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("GetBySourceId/{id:int?}")]
```

Comment:
Endpoint: `GET /documents/sources/{id}`. 
```suggestion
        [Route("sources/{id:int}")]
```